demo_server.py

```
@@ -131,15 +131,34 @@ def send_json_response(self, data):
+    import socket
+    try:
+        # Create a socket to find the local IP
+        s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+        s.connect(("8.8.8.8", 80))
+        local_ip = s.getsockname()[0]
+        s.close()
+        return local_ip
+    except Exception:
+        return "Unable to detect"
+
+def start_server(port=8000, host="0.0.0.0"):
```

Ruff (0.16.1)

[error] Possible binding to all interfaces

(S104)